client/network/src/service.rs

```diff
@@ -173,60 +180,70 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 				}
 			)?;
 
-		// Initialize the peers we should always be connected to.
-		let priority_groups = {
-			let mut reserved_nodes = HashSet::new();
-			for reserved in params.network_config.reserved_nodes.iter() {
-				reserved_nodes.insert(reserved.peer_id.clone());
-				known_addresses.push((reserved.peer_id.clone(), reserved.multiaddr.clone()));
-			}
-
-			let print_deprecated_message = match &params.role {
-				Role::Sentry { .. } => true,
-				Role::Authority { sentry_nodes } if !sentry_nodes.is_empty() => true,
-				_ => false,
-			};
-			if print_deprecated_message {
-				log::warn!(
-					"🙇 Sentry nodes are deprecated, and the `--sentry` and  `--sentry-nodes` \
-					CLI options will eventually be removed in a future version. The Substrate \
-					and Polkadot networking protocol require validators to be \
-					publicly-accessible. Please do not block access to your validator nodes. \
-					For details, see https://github.com/paritytech/substrate/issues/6845."
-				);
-			}
+		// Print a message about the deprecation of sentry nodes.
+		let print_deprecated_message = match &params.role {
+			Role::Sentry { .. } => true,
+			Role::Authority { sentry_nodes } if !sentry_nodes.is_empty() => true,
+			_ => false,
+		};
+		if print_deprecated_message {
+			log::warn!(
+				"🙇 Sentry nodes are deprecated, and the `--sentry` and  `--sentry-nodes` \
+				CLI options will eventually be removed in a future version. The Substrate \
+				and Polkadot networking protocol require validators to be \
+				publicly-accessible. Please do not block access to your validator nodes. \
+				For details, see https://github.com/paritytech/substrate/issues/6845."
+			);
+		}
 
-			let mut sentries_and_validators = HashSet::new();
-			match &params.role {
-				Role::Sentry { validators } => {
-					for validator in validators {
-						sentries_and_validators.insert(validator.peer_id.clone());
-						reserved_nodes.insert(validator.peer_id.clone());
-						known_addresses.push((validator.peer_id.clone(), validator.multiaddr.clone()));
+		let peerset_config = {
+			let mut sets = Vec::with_capacity(1 + params.network_config.extra_sets.len());
```

Comment:
Feel free to ignore as pr still in progress: I doubt pre-allocating will bring us a performance gain, given that this is only called once in the lifetime of a node and relatively small. Am I missing something?